File: Server/src/main/java/lk/wnl/wijeya/entity/MaterialGrnMaterial.java
package lk.wnl.wijeya.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "material_grn_material")
public class MaterialGrnMaterial {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Integer id;

    @Column(name = "line_cost", precision = 15, scale = 2)
    private BigDecimal lineCost;

    @Column(name = "unit_price", precision = 15, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "quantity")
    private Integer quantity;

    @JsonIgnore
    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "material_grn_id", nullable = false)
    private MaterialGrn materialGrn;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "material_id", nullable = false)
    private Material material;

}

File: Client/src/app/entity/MaterialGrnMaterial.ts
import {Material} from "./Material";

export class MaterialGrnMaterial{

  id!: number;
  material!: Material;
  quantity!: number;
  lineCost!: number;
  unitPrice!:number
  constructor(id: number, material: Material, quantity: number, lineCost: number, unitPrice: number) {
    this.id = id;
    this.material = material;
    this.quantity = quantity;
    this.lineCost = lineCost;
    this.unitPrice= unitPrice;
  }
}
